packages/medusa/src/api/routes/admin/payment-collections/refund-payment.ts

```diff
@@ -0,0 +1,40 @@
+import { IsInt, IsNotEmpty, IsOptional, IsString } from "class-validator"
+
+import { PaymentCollectionService } from "../../../../services"
+import { validator } from "../../../../utils/validator"
+
+export default async (req, res) => {
+  const { payment_id } = req.params
+
+  const validated = (await validator(
+    AdminPostPaymentCollectionRefundsReq,
+    req.body
+  )) as AdminPostPaymentCollectionRefundsReq
+
+  const paymentCollectionService: PaymentCollectionService = req.scope.resolve(
+    "paymentCollectionService"
+  )
+
+  const payments = await paymentCollectionService.refund(
```

Comment:
todo: Should be singular
```suggestion
  const payment = await paymentCollectionService.refund(
```